modules/azure/nsg.go

```diff
@@ -0,0 +1,194 @@
+		return NsgRuleSummaryList{}, err
+	}
+
+	defaultRuleList, err := defaultRulesClient.ListComplete(context.Background(), resourceGroupName, nsgName)
+	if err != nil {
+		return NsgRuleSummaryList{}, err
+	}
+
+	customRuleList, err := customRulesClient.ListComplete(context.Background(), resourceGroupName, nsgName)
+	if err != nil {
+		return NsgRuleSummaryList{}, err
+	}
+
+	rules := make([]NsgRuleSummary, 0)
+	bindRuleList(&rules, defaultRuleList)
+	bindRuleList(&rules, customRuleList)
+	ruleList := NsgRuleSummaryList{}
+	ruleList.SummarizedRules = rules
+	return ruleList, nil
+}
+
+// NsgRuleSummaryList holds a colleciton of NsgRuleSummary structs
+type NsgRuleSummaryList struct {
+	SummarizedRules []NsgRuleSummary
+}
+
+// NsgRuleSummary is a string-based summary of an NSG rule with methods attached.
+type NsgRuleSummary struct {
+	Name                     string
+	Description              string
+	Protocol                 string
+	SourcePortRange          string
+	DestinationPortRange     string
+	SourceAddressPrefix      string
+	DestinationAddressPrefix string
+	Access                   string
+	Priority                 int32
+	Direction                string
+}
+
+func bindRuleList(dest *[]NsgRuleSummary, source network.SecurityRuleListResultIterator) {
+	for source.NotDone() {
+		v := source.Value()
+		*dest = append(*dest, convertToNsgRuleSummary(v.Name, v.SecurityRulePropertiesFormat))
+		err := source.NextWithContext(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func convertToNsgRuleSummary(name *string, rule *network.SecurityRulePropertiesFormat) NsgRuleSummary {
+	summary := NsgRuleSummary{}
+
+	if rule.Description != nil {
+		summary.Description = *rule.Description
+	} else {
+		summary.Description = ""
+	}
+
+	summary.Name = *name
+	summary.Protocol = string(rule.Protocol)
+	summary.SourcePortRange = *rule.SourcePortRange
+	summary.DestinationPortRange = *rule.DestinationPortRange
+	summary.SourceAddressPrefix = *rule.SourceAddressPrefix
+	summary.DestinationAddressPrefix = *rule.DestinationAddressPrefix
+	summary.Access = string(rule.Access)
+	summary.Priority = *rule.Priority
+	summary.Direction = string(rule.Direction)
+	return summary
+}
+
+// FindRuleByName looks for a matching rule name
+func (summarizedRules *NsgRuleSummaryList) FindRuleByName(name string) NsgRuleSummary {
+	for _, r := range summarizedRules.SummarizedRules {
+		if r.Name == name {
+			return r
+		}
+	}
+
+	return NsgRuleSummary{}
+}
+
+// AllowsDestinationPort checks to see if the rule allows a specific destination port.
+func (summarizedRule *NsgRuleSummary) AllowsDestinationPort(port string) bool {
+	if summarizedRule.DestinationPortRange == "*" {
+		return true
+	}
+
+	low, high := ParsePortRangeString(summarizedRule.DestinationPortRange)
+	portAsInt, _ := strconv.ParseInt(port, 10, 16)
+
+	if (port == "*") && (low == 0) && (high == 65535) {
+		return true
+	}
+
+	return ((low <= uint16(portAsInt)) && (uint16(portAsInt) <= high))
```

Comment:
NIT: I think mentally, you expect to be comparing the port to the range, not the other way around, so it's easier to read as:

```go
portAsIntUnsigned := uint16(portAsInt)
return portAsIntUnsigned >= low && portAsIntUnsigned <= high
```

Reply:
Updated comparison order.